audit/csvformatter.go

```diff
@@ -57,33 +56,56 @@ func (f *CsvFormatter) Format(entry *Entry) ([]byte, error) {
 		var buf bytes.Buffer
 		w := csv.NewWriter(&buf)
 
-		f.write(w, []string{"Summary"})
-		f.write(w, summaryHeader)
-		f.write(w, summaryRow)
+		var err error
+		if err = f.write(w, []string{"Summary"}); err != nil {
+			return nil, err
+		}
+		if err = f.write(w, summaryHeader); err != nil {
+			return nil, err
+		}
+		if err = f.write(w, summaryRow); err != nil {
+			return nil, err
+		}
```

Comment:
Holy code duplication batperson. I'm counting 11 or so blocks that are looking almost the same. Any way to simplify this down?

Reply:
Agreed. I'm open to suggestions here. Smells funny, but I'm not seeing a less stanky approach yet.

Reply:
Maybe an anon function in the function like:

```
	writeAndCheckError = func(w *csv.Writer, value []string) {
		if err = f.write(w, value); err != nil {
			return nil, err
		}
	}
```

Haven't tested it myself but that MIGHT work. Just not sure if it will return in the function itself or return to the block where you call it.